trickuri.go

```diff
@@ -65,170 +67,178 @@ func userHomeDir() string {
-	notBefore := time.Now().AddDate(0, 0, -7) // Mitigate clock-skew between this server and the client.
-	notAfter := notBefore.AddDate(1, 0, 7)    // Certs are valid for one year from now.
-	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
-	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+// newCertificate returns an initialized certificate with usable values.
+func newCertificate() (*x509.Certificate, error) {
+	maxSerial := new(big.Int).Lsh(big.NewInt(1), 128)
+	serial, err := rand.Int(rand.Reader, maxSerial)
 	if err != nil {
 		return nil, err
 	}
 
-	certTemplate := &x509.Certificate{
-		SerialNumber: serialNumber,
+	return &x509.Certificate{
+		SerialNumber: serial,
 		Subject: pkix.Name{
 			Organization: []string{"TrickUri Interception Certificate"},
 		},
-		NotBefore:             notBefore,
-		NotAfter:              notAfter,
+		NotBefore:             time.Now().AddDate(0, 0, -7), // mitigate clock-skew
+		NotAfter:              time.Now().AddDate(1, 0, 7),
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 		BasicConstraintsValid: true,
-	}
-
-	return certTemplate, nil
+	}, nil
 }
 
-// Returns a root certificate and its key.
-func createRootCertificate() ([]byte, *rsa.PrivateKey, error) {
-	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+// newRootCertificate returns a root certificate and its key.
+func newRootCertificate() ([]byte, *rsa.PrivateKey, error) {
+	private, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	certTemplate, err := createCertificateTemplate()
+	cert, err := newCertificate()
 	if err != nil {
 		return nil, nil, err
 	}
+	cert.IsCA = true
+	cert.MaxPathLen = 0
+	cert.MaxPathLenZero = true
+	cert.Subject = pkix.Name{Organization: []string{"TrickUri Root"}}
+	hash := sha256.Sum256(private.PublicKey.N.Bytes())
+	cert.SubjectKeyId = hash[:]
+	cert.KeyUsage |= x509.KeyUsageCertSign
 
-	var cerBytes []byte
-	certTemplate.IsCA = true
-	certTemplate.MaxPathLen = 0
-	certTemplate.MaxPathLenZero = true
-	certTemplate.Subject = pkix.Name{Organization: []string{"TrickUri Root"}}
-	hash := sha256.Sum256(priv.PublicKey.N.Bytes())
-	certTemplate.SubjectKeyId = hash[:]
-	certTemplate.KeyUsage |= x509.KeyUsageCertSign
-	cerBytes, err = x509.CreateCertificate(rand.Reader, certTemplate, certTemplate, &priv.PublicKey, priv)
+	certBytes, err := x509.CreateCertificate(rand.Reader, cert, cert, &private.PublicKey, private)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return cerBytes, priv, nil
+	return certBytes, private, nil
 }
 
-// Writes |certificate| to |filename|.cer and |key| to |filename|.pem.
-func writeCertificate(certificate []byte, key *rsa.PrivateKey, filename string) error {
-	certOut, err := os.Create(*directory + "/" + filename + ".cer")
+// writeCertificate creates two files using the given contents and paths.
```

Comment:
I feel creates two files doesn't actually explain what this is meant to do, what do you think about:
"writeCertificate writes the provided certificate and key to files in the specified paths"?

Reply:
That is a better wording.